### PR TITLE
chore: maintenance of dependencies, bots, build scripts, and linting

### PR DESCRIPTION
**Pull request by mirceanis**
This PR is part of a greater effort to reduce the maintenance burden by using a standard set of scripts and tools on all repositories currently in maintenance.

Changes included here:
* switch from circleci to gh actions
  * circleci checks will show as failed until we disable the circleci build
* reschedule renovate bot for less noise
* adjust stalebot config
* revise and update dependencies and build scripts
* use eslint and fix errors
* adjust linter and compiler configs
* reformat code
* add `publicKeyMultibase` to `VerificationMethod` definition

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,46 +2,66 @@
   "name": "did-resolver",
   "version": "3.1.0",
   "description": "Resolve DID documents",
-  "main": "lib/resolver.umd.js",
   "source": "src/resolver.ts",
-  "module": "lib/resolver.modern.js",
+  "main": "lib/resolver.js",
+  "module": "lib/resolver.module.js",
```
Comment (rado0x54) [inline package.json:7]: I think now main and module keywords are correctly used (CJS / ESM)! 👍🏻

**Review by rado0x54 — CHANGES_REQUESTED** [review]

File: package.json [hunk 2]
```diff
@@ -2,46 +2,66 @@
...
   "types": "lib/resolver.d.ts",
+  "unpkg": "lib/resolver.umd.js",
+  "umd:main": "lib/resolver.umd.js",
```
[the remark this answers is not in the document]
Comment (mirceanis) [reply package.json:10]: I followed some guides from `microbundle` but I don't know when these are used, actually.
I can remove them.

**Comment by yhuard** [comment]
Hi @mirceanis! Do you plan to release a new version of the lib soon? There are changes in this PR that I would like to use :) (like the new `publicKeyMultibase` prop on VerificationMethod). Thanks!

**Comment by mirceanis** [comment]
@yhuard I just did, see version 3.1.1 :)

**Comment by yhuard** [comment]
Excellent! Thank you very much 🙇

**Pull request merged**